sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -774,9 +774,9 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
       StatsSetupConst.TOTAL_SIZE).forall(!tableMetadata.contains(_)))
 
     if (analyzedByHive) {
-      assert(StringUtils.filterPattern(describeResult1, "*numRows\\s+500*").nonEmpty)
```

Comment:
`*` will not be supported ?

Reply:
> `*` will not be supported ?

It's the problem of `\\s+` not supported

Reply:
So this PR will break change the behavior of `StringUtils.filterPattern` ?

Reply:
> So this PR will break change the behavior of `StringUtils.filterPattern` ?

We can add a little logic in `escapeLikeRegex` to solve this.